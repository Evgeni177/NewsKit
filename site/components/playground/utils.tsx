import React from 'react';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const isComponent = (value: any): value is React.ComponentType =>
  Boolean(value && (value.$$typeof || value.displayName));

export const withProps = (
  Component: React.ComponentType,
  props: Record<string, unknown>,
  displayName?: string,
) => {
  const Comp: React.FC = p => <Component {...props} {...p} />;
  Comp.displayName = displayName || Component.displayName;
  return Comp;
};
