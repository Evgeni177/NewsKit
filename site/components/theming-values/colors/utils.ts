import {Theme, ThemeProp} from 'newskit';
import {ThemeColor} from './types';
import {getContrast, getBrightness, getContrastRating} from './color-utils';

export const getBorderRadius = ({
  first,
  last,
  theme,
}: {
  first?: boolean;
  last?: boolean;
  theme?: Theme;
}) => {
  const fn = ({
    theme: {
      borderRadius: {borderRadiusRounded020: radius},
    },
  }: ThemeProp) =>
    [
      first ? radius : 0,
      first ? radius : 0,
      last ? radius : 0,
      last ? radius : 0,
    ].join(' ');

  if (theme) {
    return fn({theme});
  }

  return fn;
};

export const getTextColor = ({
  theme,
  value,
}: // brightness,
{theme: Theme} & Pick<ThemeColor, 'brightness' | 'value'>) =>
  // brightness < 0.4 ? theme.colors.inkInverse : theme.colors.inkContrast;
  getContrast(value, theme.colors.inkInverse) <
  getContrast(value, theme.colors.inkContrast)
    ? theme.colors.inkInverse
    : theme.colors.inkContrast;

export const getColorGroup = (name: string) => (/[a-z]+/.exec(name) || [])[0];

export const mapColorObjects = (
  theme: Theme,
  paletteColorMap?: Record<string, ThemeColor[]>,
) => ([name, value]: [string, string]) => {
  const brightness = getBrightness(value);
  const contrastRatio = getContrast(value, theme.colors.interface010);
  const contrastRating = getContrastRating(contrastRatio);

  return {
    name,
    value,
    group: getColorGroup(name),
    brightness,
    contrastRating,
    contrastRatio,
    parentColor:
      paletteColorMap && paletteColorMap[value]
        ? paletteColorMap[value][0]
        : undefined,
  } as ThemeColor;
};

export const groupColorObjects = (
  acc: Record<string, ThemeColor[]>,
  color: ThemeColor,
) => {
  // eslint-disable-next-line no-multi-assign
  const group = (acc[color.group] = acc[color.group] || []);
  group.push(color);
  return acc;
};

export const createPaletteColorMap = (paletteColors: ThemeColor[]) =>
  paletteColors.reduce(
    (acc, color) => {
      // eslint-disable-next-line no-multi-assign
      const accGroup = (acc[color.value] = acc[color.value] || []);
      accGroup.push(color);
      return acc;
    },
    {} as Record<string, ThemeColor[]>,
  );
