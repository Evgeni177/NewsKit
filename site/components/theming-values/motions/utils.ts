export const CANVAS_SIZE = 56;

const parseCubicBezier = (value: string) =>
  value
    .replace('cubic-bezier(', '')
    .replace(')', '')
    .split(',')
    .map(v => Number.parseFloat(v.trim()));

// NEXT 2 FUNCTIONS ARE COPIED FROM http://dbaron.org/css/timing-function-graphs
export function bezier(x1: number, y1: number, x2: number, y2: number) {
  // Cubic bezier with control points (0, 0), (x1, y1), (x2, y2), and (1, 1).
  function xForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * x1 + 3 * omt * t * t * x2 + t * t * t;
  }
  function yForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * y1 + 3 * omt * t * t * y2 + t * t * t;
  }
  function tForX(x: number) {
    // Binary subdivision.
    let mint = 0;
    let maxt = 1;
    for (let i = 0; i < 30; ++i) {
      const guesst = (mint + maxt) / 2;
      const guessx = xForT(guesst);
      if (x < guessx) maxt = guesst;
      else mint = guesst;
    }
    return (mint + maxt) / 2;
  }
  return function bezierClosure(x: number) {
    if (x === 0) return 0;
    if (x === 1) return 1;
    return yForT(tForX(x));
  };
}

export const updateCanvas = (
  ctx: CanvasRenderingContext2D,
  timingFn: string,
  color: string,
) => {
  let x1 = 0;
  let y1 = 0;
  let x2 = 0;
  let y2 = 0;
  switch (timingFn) {
    case 'ease':
      x1 = 0.25;
      y1 = 0.1;
      x2 = 0.25;
      y2 = 1.0;
      break;
    case 'linear':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-in':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-out':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    case 'ease-in-out':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    default:
      [x1, y1, x2, y2] = parseCubicBezier(timingFn);
      break;
  }

  const tf = bezier(x1, y1, x2, y2);

  const gCurrentTFPoints = [];

  for (let x = 0; x <= CANVAS_SIZE; ++x) {
    gCurrentTFPoints[x] = tf(x / CANVAS_SIZE) * CANVAS_SIZE;
  }

  ctx.clearRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);

  // Draw the graph
  ctx.strokeStyle = color;
  ctx.lineWidth = 2;
  ctx.beginPath();
  ctx.moveTo(0, CANVAS_SIZE - gCurrentTFPoints[0]);
  for (let x = 1; x <= CANVAS_SIZE; ++x) {
    ctx.lineTo(x, CANVAS_SIZE - gCurrentTFPoints[x]);
  }
  ctx.stroke();
};

export const percentFrom = (total: number, value: number) =>
  Math.floor(value / (total / 100));

export const getTotalDuration = (stayDuration: number, maxDuration: number) =>
  2 * stayDuration + 2 * maxDuration;
