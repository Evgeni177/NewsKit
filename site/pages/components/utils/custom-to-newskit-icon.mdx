import {LegacyBlock} from '../../../components/legacy-block';
import Layout from '../../../components/layout';
import {Code, CodeFromFile} from '../../../components/code';
import Prop from '../../../components/prop';

export default Layout;

# customToNewsKitIcon()

## Syntax

<Code>{'customToNewsKitIcon(displayName, CustomIcon, overrides)'}</Code>

## Overview

Transform SVG markup into a Newskit icon that works well with theme tokens and `getFromTheme` util functions.

## Parameters

<LegacyBlock marginTop="sizing040">
  <Prop name="displayName" type="string" required>
    The display name of the icon. It is required an icon name to start either
    with <b>`IconFilled`</b> or <b>`IconOutlined`</b> prefix, depending on the
    imported icon type. <br />
    E.g: If importing an <b>MenuBook</b> icon, the final icon display name should
    be <b>`IconFilledMenuBook`</b> or <b>`IconOutlinedMenuBook`</b>
  </Prop>
  <Prop name="CustomIcon" type="React.ComponentType<SvgProps>" required>
    NewsKit Svg component with SVG passed to it's children, props should be
    spread on the root Svg component, see example below.
  </Prop>
  <Prop name="overrides" type='SvgProps["overrides"]'>
    Default <a href="/components/icons#props">override props</a> passed to
    the transformed icon. Overrides can still be passed to the icon to overwrite
    these values at render time.
  </Prop>
</LegacyBlock>

## Return value

Returns a NewsKit icon.

## Example

<CodeFromFile path="examples/utils/custom-to-newskit-icon.tsx" />
