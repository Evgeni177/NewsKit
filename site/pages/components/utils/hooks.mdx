import Layout from '../../../components/layout';
import {Code} from '../../../components/code';
import {LegacyBlock} from '../../../components/legacy-block';
import Prop from '../../../components/prop';
import {Link} from 'newskit';

export default Layout;

# Hooks

## useMediaQueryObject

### Overview

`useMediaQueryObject` hook handles scenarios in which you want to render component based on media query breakpoints.
This hooks also responds to the window resizing and returns the appropriate value for the new window size.

`useMediaQueryObject` returns the value for the current breakpoint from the provided media query object ( `MQ<T>` )

Note: The hook needs `<MediaQueryProvider>` to be added to an ancestor of the component in which it's used.

### Example

A common use case for using `useMediaQueryObject` is when you want to change component property based on media query.
In the example below, the `<Card />` component changes its layout depending on screen the size.

<Code>
  {`const mqLayout = {
    xs: 'vertical',
    sm: 'vertical',
    md: 'horizontal',
    lg: 'horizontal',
    xl: 'horizontal',
}
const layout = useMediaQueryObject(mqLayout);
<Card layout={layout} /> `}
</Code>

Note: In theory `useMediaQueryObject` could be used with any `MQ<T>`, however, we recommend using it only for props that don't accept `MQ<T>` as a value. Most CSS-based props and overrides already support `MQ<T>` objects.
When you create your own components and want to apply responsive behaviour you should try to use <Link href="/utils/get-from-theme">getters</Link> and <Link href="/utils/get-defaults">defaults</Link> functions.

DO NOT DO THIS:

<Code>
  {`const stylePresets = {
    xs: 'dividerPrimary',
    md: 'horizontal',
}
const dividerStylePreset = useMediaQueryObject(stylePresets);
<Divider overrides={{stylePreset: dividerStylePreset}} />`}
</Code>

## useBreakpointKey

### Overview

`useBreakpointKey` has a similar utility as `useMediaQueryObject`, it's intended usage is where you want to know the currently active breakpoint key `xs | sm | md | lg | xl`.

### Example

<Code>
  {`const breakpointKey = useBreakpointKey();
// return XL when screen has width XL ( 1440px and above )`}
</Code>

`<Card />` component use case

<Code>
  {`const breakpointKey = useBreakpointKey();
  const layout == breakpointKey === 'xs || breakpointKey === 'sm' ? 'vertical' : 'horizontal';
<Card layout={layout} />
// the <Card will render verticaly on XS and SM screens and Horizontaly on the rest of breakpoints`}

</Code>
