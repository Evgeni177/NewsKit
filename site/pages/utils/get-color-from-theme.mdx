import {LegacyBlock} from '../../components/legacy-block';
import Layout from '../../components/layout';
import {Code, CodeFromFile} from '../../components/code';
import Prop from '../../components/prop';

export default Layout;

# getColorFromTheme()

## Syntax

<Code>{'getColorFromTheme(defaultToken, customProp)(props)'}</Code>

## Overview

Retrieve a color token from the current theme. Color tokens are [CSS color values](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) to be used for CSS styling.

## Parameters

<LegacyBlock marginTop='sizing040'>
  <Prop name='defaultToken' type='ThemeToken'>
    Is the name of the color token to be retrieved. If provided, it is used as an object key on the color section of the theme to get the token value.
  </Prop>
  <Prop name='customProp' type="Exclude<keyof Props, 'theme'>">
    Is the name of the component prop in which a color token name can be provided by the end-user of the component. If provided, the color to be retrieved will come from the component prop.
  </Prop>
  <Prop name='props' type='Props'>
    Is the props of the component including the theme. If the function is used inside a styled component template literal, or a css block, Emotion.js will call the function first with the component props passed in.
  </Prop>
</LegacyBlock>

## Return value

Returns a color token if valid attributes are received.

Returns an empty string if invalid attributes or no attributes are provided.

## Example

<CodeFromFile path="examples/utils/get-color-from-theme.tsx"/>

In this example the `Paragraph` component will include the color value from `highlight` found in the theme if the `Paragraph` is used with no props:

<Code>{'<Paragraph />'}</Code>

However, it will include value from `focus` if the `Paragraph` is used with the `color` prop and the value passed in is `focus`:

<Code>{'<Paragraph color="focus" />'}</Code>
