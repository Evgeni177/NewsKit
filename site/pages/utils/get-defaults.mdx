import {LegacyBlock} from '../../components/legacy-block';
import Layout from '../../components/layout';
import {Code, CodeFromFile} from '../../components/code';
import Prop from '../../components/prop';
import {Image} from 'newskit';
import {Link, LinkStandalone} from 'newskit';

export default Layout;

# Component Defaults Utils

A group of functions used to retrieve values from the <Link href="/theming/component-defaults">component defaults or overrides objects.</Link>

## Overview

Each component has its own component defaults object as well as overrides prop.
Component defaults map foundation and preset tokens to the component, describing the default look and feel. As well as being able to override these defaults in the theme, component overrides allow you to change these tokens as desired when rendering individual instances.<br/>

The provided functions, specified below, are used to retrieve the CSS values and objects sitting behind the theme tokens. These functions don't access the theme directly themselves, they just orchestrate the process of fetching theme values, utilising the other _get from theme_ functions.

## Parameters

<LegacyBlock marginTop='sizing040'>
  <Prop name='defaultPath' type='string | undefined'>
    It is an object-path used to access deep properties in the component defaults. <br/>
    Component defaults objects can have a nested structure that mirrors the structure of the component. Root level key of this object is always the name of the component. <br/>
    Example: "headline.kicker", "byline.divider", "shareBar"
    If not provided (undefined or false passed) desired value is looked up only in the respective overrides object.
  </Prop>
  <Prop name='overridePath' type='string | false' default='""'>
    It is an object-path used to access deep properties in the overrides. <br/>
    Overrides objects can have as well a nested structure that mirrors the structure of the component. <br/>
    Example: "kicker", "divider", "" (empty string)
  </Prop>
  <Prop name='options' type='Parameters<FromThemeUtil>[2]'>
    This prop consists of different options used to control component-specific behaviours.
  </Prop>
  <Prop name='props' type='Props'>
    Is the props of the component including the theme. If the function is used inside a styled component template literal, or a CSS block, Emotion.js will call the function first with the component props passed in.
  </Prop>
</LegacyBlock>

## Syntax
<Code>{
  `
  getStylePreset(defaultPath, overridePath, options)(props)
  
  options: {
    isLoading?: boolean;
    isSelected?: boolean;
    isDisabled?: boolean;
    isInvalid?: boolean;
    isValid?: boolean;
    isSvg?: boolean;
    omitStates?: StylePresetStates[];
    filterStates?: StylePresetStates[];
    omitStyles?: StylePresetStyleKeys[];
    filterStyles?: StylePresetStyleKeys[];
  };
  `
}</Code>

Used to retrieve values mapped to the stylePreset key in the component defaults/overrides.

<Code>{
  `
  getTypographyPreset(defaultPath, overridePath, options)(props)
  
  options: {
    withCrop?: boolean;
  };
  `
}</Code>

Used to retrieve values mapped to the typographyPreset key in the component defaults/overrides.

<Code>{'getSpacingInlineHorizontal(defaultPath, overridePath, options)(props)'}</Code>

Used to retrieve value mapped to the <Link href="/theming/foundations/spacing">spaceInline</Link> key in the component defaults/overrides 
and places it as margin-right CSS property when the text or itmes flow is horizontal.

<Code>{'getSpacingInlineVertical(defaultPath, overridePath, options)(props)'}</Code>

Used to retrieve value mapped to the <Link href="/theming/foundations/spacing">spaceInline</Link> key in the component defaults/overrides 
and places it as margin-bottom CSS property when the text or itmes flow is vertical.

<Code>{'getSpacingStackHorizontal(defaultPath, overridePath, options)(props)'}</Code>

Used to retrieve value mapped to the <Link href="/theming/foundations/spacing">spaceStack</Link> key in the component defaults/overrides 
and places it as margin-bottom CSS property when the text or itmes flow is horizontal.

<Code>{'getSpacingStackVertical(defaultPath, overridePath, options)(props)'}</Code>

Used to retrieve value mapped to the <Link href="/theming/foundations/spacing">spaceStack</Link> key in the component defaults/overrides 
and places it as margin-right CSS property when the text or itmes flow is vertical.

<Code>{'getSpacingInset(defaultPath, overridePath, options)(props)'}</Code>

Used to retrieve value mapped to the spaceInsite key in the component defaults/overrides 
and applies it as padding CSS property to an element.


## Return value

Returns a CSS value, or object, mapped to a theme token (mapped to a key in the component defaults/overrides) if valid attributes are received.

Returns an empty string if invalid attributes or no attributes are provided.

## Example

Here is an example of using the different functions:

<CodeFromFile path="examples/utils/get-defaults.tsx"/>

