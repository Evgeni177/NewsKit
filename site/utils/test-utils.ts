import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
