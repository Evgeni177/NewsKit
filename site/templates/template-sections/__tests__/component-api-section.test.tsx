import React from 'react';
import {ComponentAPISection, ComponentAPISectionProps} from '..';
import {renderToFragmentWithTheme} from '../../../utils/test-utils';

jest.mock(
  'newskit',
  require('../../../utils/test-utils').mockNewsKitComponents(
    'Grid',
    'Cell',
    'Tabs',
    'Tab',
  ),
);

jest.mock('../../../components/table/table');

describe('ComponentAPISection', () => {
  test('renders section as expected', () => {
    const props: ComponentAPISectionProps = {
      introduction: 'component api introduction text goes here',
      components: [
        {
          title: 'Component',
          summary: 'Summary for the component variant',
          propsRows: [
            {
              name: 'children',
              type: 'string',
              required: true,
              description: (
                <>The content of the Component is passed as the children.</>
              ),
            },
          ],
          overridesRows: [
            {
              attribute: 'spaceInset',
              type: 'MQ<string>',
              description:
                'If provided, this overrides the padding of the Component.',
            },
          ],
        },
      ],
    };

    const fragment = renderToFragmentWithTheme(ComponentAPISection, props);
    expect(fragment).toMatchSnapshot();
  });
});
