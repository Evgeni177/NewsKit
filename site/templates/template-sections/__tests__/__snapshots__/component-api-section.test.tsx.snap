// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentAPISection renders section as expected 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-top: -24px;
  padding-top: 24px;
}

@media screen and (max-width: 767px) {
  .emotion-1 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-1 {
    margin-bottom: 32px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0;
  background-color: transparent;
}

.emotion-4 {
  padding: 1px 0px;
}

@media screen and (max-width: 767px) {
  .emotion-4 {
    font-family: "Poppins",sans-serif;
    font-size: 24px;
    line-height: 1.1666666666666667;
    font-weight: 700;
    letter-spacing: -0.5px;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.2645833333333334em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.2645833333333334em;
  }
}

@media screen and (min-width: 768px) {
  .emotion-4 {
    font-family: "Poppins",sans-serif;
    font-size: 28px;
    line-height: 1.1428571428571428;
    font-weight: 700;
    letter-spacing: -0.5px;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.2526785714285714em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.2526785714285714em;
  }
}

.emotion-5 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0A0A0A;
}

.emotion-5 svg {
  fill: #0A0A0A;
}

@media screen and (max-width: 1023px) {
  .emotion-6 {
    margin-bottom: 48px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-6 {
    margin-bottom: 80px;
  }
}

.emotion-7 {
  margin: 0;
  padding: 1px 0;
  color: #2E2E2E;
  font-family: "Poppins",sans-serif;
  font-size: 20px;
  line-height: 1.6;
  font-weight: 500;
  letter-spacing: -0.5px;
}

.emotion-7 svg {
  fill: #2E2E2E;
}

.emotion-7::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.48125000000000007em;
}

.emotion-7::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.48125000000000007em;
}

@media screen and (max-width: 767px) {
  .emotion-9 {
    margin-bottom: 16px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-9 {
    margin-bottom: 20px;
  }
}

.emotion-10 {
  margin: 0;
  padding: 1px 0;
  color: #0A0A0A;
  display: inline-block;
}

.emotion-10 svg {
  fill: #0A0A0A;
}

@media screen and (max-width: 1439px) {
  .emotion-10 {
    font-family: "Poppins",sans-serif;
    font-size: 20px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: -0.5px;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.28125em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.28125em;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-10 {
    font-family: "Poppins",sans-serif;
    font-size: 22px;
    line-height: 1.0909090909090908;
    font-weight: 700;
    letter-spacing: -0.5px;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.22670454545454544em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.22670454545454544em;
  }
}

.emotion-11 {
  margin-bottom: 40px;
}

.emotion-12 {
  margin: 0;
  padding: 1px 0;
  color: #2E2E2E;
}

.emotion-12 svg {
  fill: #2E2E2E;
}

@media screen and (max-width: 1439px) {
  .emotion-12 {
    font-family: "DM Sans",sans-serif;
    font-size: 16px;
    line-height: 1.5;
    font-weight: 400;
    letter-spacing: -0.5px;
  }

  .emotion-12::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.43125em;
  }

  .emotion-12::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.43125em;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-12 {
    font-family: "DM Sans",sans-serif;
    font-size: 18px;
    line-height: 1.5555555555555556;
    font-weight: 400;
    letter-spacing: -0.5px;
  }

  .emotion-12::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.4590277777777778em;
  }

  .emotion-12::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.4590277777777778em;
  }
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12
}"
    data-testid="Cell"
  >
    <section
      class="emotion-0"
      data-toc-indexed="API"
      id="component-api"
    >
      <div
        data-comp="NewsKit Grid"
        data-props="{
  \\"xsMargin\\": \\"space000\\",
  \\"xsRowGutter\\": \\"space000\\"
}"
        data-testid="Grid"
      >
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 8,
  \\"lg\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <section
                  class="emotion-4"
                >
                  <h2
                    class="nk-headline-heading emotion-5"
                  >
                    API
                  </h2>
                </section>
              </div>
            </div>
          </div>
          <div
            class="emotion-6"
          >
            <p
              class="emotion-7"
            >
              component api introduction text goes here
            </p>
          </div>
        </div>
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"mdOffset\\": 1,
  \\"md\\": 10,
  \\"xl\\": 8
}"
          data-testid="Cell"
        />
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"lg\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div
            class="emotion-3"
          >
            <div
              class="emotion-9"
            >
              <span
                class="emotion-10"
              >
                Component
              </span>
            </div>
            <div
              class="emotion-11"
            >
              <p
                class="emotion-12"
              >
                Summary for the component variant
              </p>
            </div>
            <div
              data-comp="NewsKit Tabs"
              data-props="{
  \\"size\\": \\"medium\\"
}"
              data-testid="Tabs"
            >
              <div
                data-comp="NewsKit Tab"
                data-props="{
  \\"label\\": \\"Props\\",
  \\"overrides\\": {
    \\"typographyPreset\\": \\"utilityButton020\\",
    \\"stylePreset\\": \\"componentPageTabs\\"
  }
}"
                data-testid="Tab"
              >
                <div>
                  Table Component with props:{
  "columns": [
    "Name",
    "Type",
    "Default",
    "Description",
    "Required"
  ],
  "rows": [
    {
      "name": "children",
      "type": "string",
      "required": true,
      "description": {
        "key": null,
        "ref": null,
        "props": {
          "children": "The content of the Component is passed as the children."
        },
        "_owner": null,
        "_store": {}
      }
    }
  ]
}
                </div>
              </div>
              <div
                data-comp="NewsKit Tab"
                data-props="{
  \\"label\\": \\"Overrides\\",
  \\"overrides\\": {
    \\"typographyPreset\\": \\"utilityButton020\\",
    \\"stylePreset\\": \\"componentPageTabs\\"
  }
}"
                data-testid="Tab"
              >
                <div>
                  Table Component with props:{
  "columns": [
    "Attribute",
    "Type",
    "Default",
    "Description"
  ],
  "rows": [
    {
      "attribute": "spaceInset",
      "type": "MQ&lt;string&gt;",
      "description": "If provided, this overrides the padding of the Component."
    }
  ]
}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    margin-top: 64px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-top: 80px;
  }
}

@media screen and (max-width: 767px) {
  .emotion-0 {
    margin-bottom: 40px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-bottom: 80px;
  }
}

.emotion-1 {
  border-style: solid;
  border-color: #E4E4E4;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-top-width: 1px;
  width: 100%;
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"lg\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
    data-testid="Cell"
  >
    <div
      class="emotion-0"
    >
      <hr
        aria-hidden="true"
        class="emotion-1"
        data-testid="divider"
      />
    </div>
  </div>
</DocumentFragment>
`;
